docker/build-env-release.Dockerfile

```diff
@@ -0,0 +1,137 @@
+FROM ubuntu:18.04
+
+WORKDIR /root
+
+SHELL ["/bin/bash", "-l", "-c"]
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV FORCE_UNSAFE_CONFIGURE 1
+
+# Install basic tools
+RUN apt-get update -qq && apt-get install -qq --no-install-recommends \
+    software-properties-common \
+    build-essential \
+    git tar wget curl gpg-agent && \
+    rm -rf /var/lib/apt/lists/*
+
+# Install cmake
+RUN wget -qO- "https://cmake.org/files/v3.17/cmake-3.17.0-Linux-x86_64.tar.gz" | tar --strip-components=1 -xz -C /usr/local
+
+# Install MPICH ABI compatible with Cray's lib on Piz Daint
+ARG MPICH_VERSION=3.3.2
+ARG MPICH_PATH=/usr/local/mpich
+RUN wget -q https://www.mpich.org/static/downloads/${MPICH_VERSION}/mpich-${MPICH_VERSION}.tar.gz && \
+    tar -xzf mpich-${MPICH_VERSION}.tar.gz && \
+    cd mpich-${MPICH_VERSION} && \
+    ./configure \
+      --disable-fortran \
+      --prefix=$MPICH_PATH && \
+    make install -j$(nproc) && \
+    rm -rf /root/mpich-${MPICH_VERSION}.tar.gz /root/mpich-${MPICH_VERSION}
+
+# Install MKL
+ARG MKL_VERSION=2020.0-088
+RUN wget -qO - https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS-2019.PUB 2>/dev/null | apt-key add - && \
+    apt-add-repository 'deb https://apt.repos.intel.com/mkl all main' && \
+    apt-get install -y -qq --no-install-recommends intel-mkl-64bit-${MKL_VERSION} && \
+    rm -rf /var/lib/apt/lists/* && \
+    echo "/opt/intel/lib/intel64\n/opt/intel/compilers_and_libraries/linux/mkl/lib/intel64" >> /etc/ld.so.conf.d/intel.conf && \
+    ldconfig
```

Comment:
I see that in release you use MKL, while in debug OpenBLAS. Is it different on purpose?

Reply:
I was thinking that you would have debug symbols for all dependencies now, even BLAS. With MKL I don't think you get them (or they would not be of any use anyways because it's closed source).

Reply:
Good point! I don't know what @rasolca thinks about this, but if one side it is very nice to have two different dependencies built and tested (MKL and OpenBLAS), at the same time I'm wondering if it wouldn't be better to build the same configuration in debug and release.

Reply:
At some point we should define the matrix of the configurations we want to test (maybe a smaller matrix for PRs and a bigger one for weekly(?) tests), but it is not responsibility of @haampie.